src/udev/udevd.c

```diff
@@ -363,13 +368,28 @@ static void device_broadcast(sd_device_monitor *monitor, sd_device *dev) {
         if (!monitor)
                 return;
 
+        if (result != 0) {
+                assert(result != EVENT_RESULT_TRY_AGAIN);
+
+                (void) device_add_property(dev, "UDEV_WORKER_FAILED", "1");
+                if (result < 0) {
+                        (void) device_add_propertyf(dev, "UDEV_WORKER_ERRNO", "%i", -result);
+                        (void) device_add_propertyf(dev, "UDEV_WORKER_ERRNO_NAME", "%s", strna(errno_to_name(result)));
+                } else if (result <= EVENT_RESULT_EXIT_STATUS_MAX)
+                        (void) device_add_propertyf(dev, "UDEV_WORKER_EXIT_STATUS", "%i", result);
+                else {
+                        (void) device_add_propertyf(dev, "UDEV_WORKER_SIGNAL", "%i", result - EVENT_RESULT_SIGNAL_BASE);
+                        (void) device_add_propertyf(dev, "UDEV_WORKER_SIGNAL_NAME", "%s", strna(signal_to_string(result - EVENT_RESULT_SIGNAL_BASE)));
+                }
```

Comment:
doesn't matter at all, but this would be super pretty with gcc range switch statements, i.e.:

```c
switch(result)

case EVENT_RESULT_NERRNO_MIN..EVENT_RESULT_NERRNO_MAX:
        …
case EVENT_RESULT_EXIT_STATUS_BASE..EVENT_RESULT_EXIT_STATUS_MAX:
        …
case EVENT_RESULT_SIGNAL_BASE..EVENT_RESULT_SIGNAL_MAX:
        …
default:
        assert_not_reached();
}
```
or so.